datafusion/functions/src/macros.rs

```diff
@@ -59,6 +59,24 @@ macro_rules! export_functions {
         }
     };
 
+    // function that requires config and takes a vector argument
+    (single $FUNC:ident, $DOC:expr, @config $arg:ident,) => {
+        #[doc = $DOC]
+        pub fn $FUNC($arg: Vec<datafusion_expr::Expr>) -> datafusion_expr::Expr {
+            use datafusion_common::config::ConfigOptions;
+            super::$FUNC(&ConfigOptions::default()).call($arg)
```

Comment:
Can someone remind me why we are going through all the trouble of threading the ConfigOptions down into the functions, when the code to create the functions just passes in the default ConfigOptions?

Shouldn't the current ConfigOptions be passed in? If so perhaps we can do this as a follow on PR